RepositoryCompiler/CodeModel/CodeParsers/CSharp/CSharpMetricCalculator.cs

```diff
@@ -40,41 +40,43 @@ public CaDETClassMetrics CalculateClassMetrics(CaDETClass parsedClass)
         private double? GetTightClassCohesion(CaDETClass parsedClass)
         {
             int N = GetNumberOfMethodsDeclared(parsedClass);
+           
             double NP = (N * (N - 1)) / 2;
             if (NP == 0) return null;
 
-            return NumberOfDirectConnections(parsedClass) / NP;
+            return GetNumberOfDirectConnections(parsedClass) / NP;
         }
 
-        private int NumberOfDirectConnections(CaDETClass parsedClass)
+        private int GetNumberOfDirectConnections(CaDETClass parsedClass)
         {
-            int mapIndex = 1;
-            var map = new Dictionary<int, List<CaDETMember>>();
+            var ownAccessedFieldsAndAccessors = new Dictionary<string, List<CaDETMember>>();
 
             foreach (var method in parsedClass.Methods)
             {
-                map.Add(mapIndex, OwnAccessedFieldsAndAccessors(method));
-                mapIndex++;
+                ownAccessedFieldsAndAccessors.Add(method.Name, OwnAccessedFieldsAndAccessors(method));
             }
 
-            int countDirectConnections = 0;
+            return CountNumberOfDirectConnections(ownAccessedFieldsAndAccessors);
```

Comment:
CountNumberOfDirectConnections can be simplified to CountDirectConnections (counting something implies we are examining a number of something).
However, I would also argue that this is a mysterious name as it is unclea what a direct connection is. Maybe something like "CountMethodPairsThatShareAccessToAFieldOrAccessor"?

Reply:
I like the suggestion, I just have dilemma whether such a long name will contribute to the general incomprehensibility of the code?

Reply:
I agree, but I can't come up with a shorter name at the moment. In general, I would err on the side of having too expressive names (i.e., many words) over mysterious names (e.g., few words or generic words).